### ISSUE TITLE
Layout issue with titles

### ISSUE THREAD

**Issue by achimha** [madoar/ng2-archwizard#4]
The steps indicator should probably provide support for situations where the screen is not wide enough to fit all step titles.

<img width="433" alt="screen shot 2017-05-27 at 16 25 51" src="https://cloud.githubusercontent.com/assets/6816140/26522090/42e72158-42f9-11e7-8367-9171fd6b2bce.png">

Any suggestions how to do this?

**Comment by madoar** [comment]
Internally `ng2-archwizard` uses `<a>` elements for the step titles. Therefore I would try to add [zero width spaces](https://stackoverflow.com/questions/15841109/optional-line-breaking-html-entity-that-is-always-invisible) at positions where you think a newline would fit in. Hopefully your browser will then automatically do a newline if the space is not enough instead of overlapping the step titles.

Other, not so beautiful approaches I can think of are:
- setting a minimum width for the wizard. I know this would lead to a scrollbar, which is not really nice to have
- use `left` or `right` for the `navBarLocation`. This isn't really a fix, but in some cases it helps if your main step content can be better compressed than the step title

If you find a better solution, please don't hesitate to tell me.

**Comment by achimha** [comment]
Unfortunately the zero width space doesn't appear to work.

Another approach would be to use different labels at different screen sizes, e.g. through bootstrap classes.

We'd have more flexibility if the step title was not just given as a string to the step but could be an `<ng-template>` as well with custom rendering. Like this:

```
<wizard-step>
  <ng-template wizardStepTitle>
    <span class="hidden-sm-down">COMMUNICATION</span>
    <span class="hidden-md-up">COMM</span>
  </ng-template>
</wizard-step>
```

**Comment by madoar** [comment]
Are you sure the zero width spaces don't work?
A few moments ago I tried them with the demo project and they appear to work.
To make them work you need to use numeric character reference format (the same as for the `navigationSymbol`):
```html
...
<wizard-step title="Del&#8203;iver&#8203;y addr&#8203;ess" navigationSymbol='&#xf1ba;' navigationSymbolFontFamily="FontAwesome">
...
```
Shown in full width, when the space is enough:
![grafik](https://cloud.githubusercontent.com/assets/18488086/26522628/ad71f4ca-4304-11e7-8c27-08bf59834511.png)
Shown with line breaks, when the space is not enough:
![grafik](https://cloud.githubusercontent.com/assets/18488086/26522629/af2326b8-4304-11e7-84cb-71623aa4c631.png)

**Comment by achimha** [comment]
You're right, it does work. Sorry about that.

Still, I see a strong use case for having different titles depending on screen size for true responsive design. The template reference approach is quite common in Angular land. Do you think that would be possible?

**Comment by madoar** [comment]
I've taken a look at the `ng-template` tag at https://angular.io/docs/ts/latest/cookbook/dynamic-component-loader.html 
I would like to give the developer more freedom to define the step titles in the navigation bar. The only problem I see, for which I know solution, is how to get the defined `wizard-step-title` from inside the `wizard-step` components into the `wizard-navigation-bar` component. Do you know a solution for this?

**Comment by achimha** [comment]
You can define a directive for the template:
```
@Directive({selector: 'ng-template[wizardStepTitle]'})
export class WizardStepTitle {
  constructor(public templateRef: TemplateRef<any>) {}
}
```
then you can capture it in the `wizard-step`:
```
@ContentChild(WizardStepTitle) titleTpl: WizardStepTitle;
```
which defines the property `titleTpl` just like the property `title` you already have.

**Comment by madoar** [comment]
I know this part. 
The part I don't know is how to insert `titleTpl` into the `wizard-navigation-bar`.
I guess it's not as easy as inserting `{{ step.titleTpl }}`, right?

**Comment by achimha** [comment]
You can do
```
{{step.title}}<ng-template [ngTemplateOutlet]="step.titleTpl?.templateRef"></ng-template>
```
Either the `title` or the `titleTpl` property will be defined.

**Comment by madoar** [comment]
This looks interesting, I'll take a look at this.

**Comment by madoar** [comment]
I've just (yesterday) created a new branch (v.1.6.0) for the title templating feature.
It would be nice if you can take a look at it and maybe test it if it satisfies your requirements and complies to what you had in mind.
The link to the branch is https://github.com/madoar/ng2-archwizard/tree/v1.6.0.

**Comment by achimha** [comment]
It looks really nice, thank you very much!

**Comment by madoar** [comment]
I've just released version 1.6.0. If you have any other ideas or problems with `ng2-archwizard`, please don't hesitate to post them.
